### PR TITLE
Extract metadata

### PR DESCRIPTION
**Pull request by mauromorales**

### CODE REVIEW

File: machinery-helper/machinery_helper.go [hunk 1]
```diff
@@ -236,11 +263,86 @@ func findUnmanagedFiles(dir string, rpmFiles map[string]string, rpmDirs map[stri
...
 }
 
+func amendMode(entry *UnmanagedFile, perm os.FileMode) {
+	if entry.Type == "link" {
+		return
+	}
+
+	result := int64(perm.Perm())
+
+	if perm&os.ModeSticky > 0 {
```
Comment (mschnitzer) [inline machinery-helper/machinery_helper.go]: I would add a space here before and after the `&`.

File: machinery-helper/machinery_helper.go [hunk 2]
```diff
@@ -236,11 +263,86 @@ func findUnmanagedFiles(dir string, rpmFiles map[string]string, rpmDirs map[stri
...
+		result |= 01000
+	}
+	if perm&os.ModeSetuid > 0 {
```
Comment (mschnitzer) [inline machinery-helper/machinery_helper.go]: I would add a space here before and after the `&`.

File: machinery-helper/machinery_helper.go [hunk 3]
```diff
@@ -236,11 +263,86 @@ func findUnmanagedFiles(dir string, rpmFiles map[string]string, rpmDirs map[stri
...
+		result |= 04000
+	}
+	if perm&os.ModeSetgid > 0 {
```
Comment (mschnitzer) [inline machinery-helper/machinery_helper.go]: I would add a space here before and after the `&`.

File: machinery-helper/machinery_helper.go [hunk 4]
```diff
@@ -236,11 +263,86 @@ func findUnmanagedFiles(dir string, rpmFiles map[string]string, rpmDirs map[stri
...
+	size = int64(0)
+	fileCount = len(files)
+	for _, f := range files {
+		if f.IsDir() {
+			if _, ok := IgnoreList[path + f.Name()]; !ok {
+				subSize, subCount := dirInfo(path + f.Name() + "/")
+				size += subSize
+				fileCount += subCount
+			}
+		} else if f.Mode()&os.ModeSymlink != os.ModeSymlink {
```
Comment (mschnitzer) [inline machinery-helper/machinery_helper.go]: I would add a space here before and after the `&`.

File: schema/migrations/migrate6to7.rb [hunk 5]
```diff
@@ -0,0 +1,38 @@
+# Copyright (c) 2013-2014 SUSE LLC
```
Comment (mschnitzer) [inline schema/migrations/migrate6to7.rb]: Adjust the year to 2016 please

File: spec/unit/migrations/migrate6to7_spec.rb [hunk 6]
```diff
@@ -0,0 +1,66 @@
+# Copyright (c) 2013-2014 SUSE LLC
```
Comment (mschnitzer) [inline spec/unit/migrations/migrate6to7_spec.rb]: Please adjust the year to 2016.

File: machinery-helper/machinery_helper.go (hunk 4 above)
Comment (mauromorales) [inline machinery-helper/machinery_helper.go]: I prefer the way it is right now and go fmt doesn't have a preference for this and go sources like the os package use it without the spaces when doing a validation like in

```
func (m FileMode) IsRegular() bool {
	return m&ModeType == 0
}
```
https://golang.org/src/os/types.go?s=3244:3278#L83

They also seem to use it the other way around when just doing the bit operation like in https://golang.org/src/os/types.go?s=3355:3388#L88

We could start defining this kind of thing on a

File: machinery-helper/file_utils_test.go [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+	"testing"
+)
+
+func TestGetFileOwnerGroup(t *testing.T) {
+	statFile = func(path string) string {
+		return "foo:bar"
+	}
+	path := "/etc/passwd"
+
+	owner, group := getFileOwnerGroup(path)
```
Comment (mschnitzer) [inline machinery-helper/file_utils_test.go]: I would do the variable assignment in one "block" like:
```go
func TestGetFileOwnerGroup(t *testing.T) {
	statFile = func(path string) string {
		return "foo:bar"
	}

	path := "/etc/passwd"
	owner, group := getFileOwnerGroup(path)

    if owner != "foo" {
```

File: machinery-helper/machinery_helper.go (hunk 4 above)
Comment (mschnitzer) [inline machinery-helper/machinery_helper.go]: OK, I'm fine with that.

**Pull request closed without merging**